pkg/krmtotf/templating.go

```diff
@@ -182,6 +182,16 @@ func expandFieldTemplate(template string, r *Resource, c client.Client, smLoader
 		if val, exists, _ := unstructured.NestedString(r.GetStatusOrObservedState(), strings.Split(path, ".")...); exists {
 			return val
 		}
+
+		// todo(yuhou): Special handling to resolve project from DCL-based resource
+		// Only used for fixtures/globalcomputebackendservicesecuritysettings test for now
+		if path == "project" {
```

Comment:
I won't worry about this because this is the only case for now where we try to resolve a reference DCL resource using TF controller. So it won't fall in previous `if` conditions and will ultimately be handled here. But I'll add a condition to make it clearer.